modules/icgc-argo-workflows/checkinput/resources/usr/bin/dnaaln.py

```diff
@@ -0,0 +1,483 @@
+        self._validate_fastq_1(row)
+        self._validate_fastq_2(row)
+        self._validate_library_name(row)
+        self._validate_platform_unit(row)
+        self._validate_platform_col(row) if row.get(self._platform_col) else ""
+        self._validate_sequencing_center_col(row) if row.get(self._sequencing_center_col) else ""
+        self._validate_sequencing_date_col(row) if row.get(self._sequencing_date_col) else ""
+        self._validate_platform_model_col(row) if row.get(self._platform_model_col) else ""
+        self._validate_read_group_count(row)
+        self._validate_experiment(row) if row.get(self._experiment_col) else ""
+        self._validate_analysis_json(row) if row.get(self._analysis_json_col) else ""
+
+        tmp_dict={
+            "analysis_type" : row[self._analysis_type_col] if row.get(self._analysis_type_col) else None,
+            "study_id" : row[self._study_id_col] if row.get(self._study_id_col) else "LOCAL",
+            "patient" : row[self._patient_col] if row.get(self._patient_col) else row[self._sample_col],
+            "sex" : row[self._sex_col] if row.get(self._sex_col) else "NA",
+            "status" : row[self._status_col] if row.get(self._status_col) else "0",
+            "sample" : row[self._sample_col],
+            "lane" : row[self._lane_col],
+            "fastq_1" : row[self._fastq_1_col],
+            "fastq_2" : row[self._fastq_2_col],
+            "single_end" : row[self._single_end_col],
+            "read_group_count" : row[self._read_group_count_col],
+            "experiment" : row[self._experiment_col] if row.get(self._experiment_col) else "WGS",
+            "analysis_json": row[self._analysis_json_col] if row.get(self._analysis_json_col) else None
+            }
+
+        read_group_info=[]
+        description=[]
+
+        for col in [
+            'experiment',
+            'study_id',
+            'experiment',
+            'patient',
+            'sample',
+            'status'
+        ]:
+            if tmp_dict.get(col):
+                if col=='status':
+                    if tmp_dict['status']==1:
+                        description.append("Tumour")
+                    else:
+                        description.append("Normal")
+                        continue
+                description.append(tmp_dict[col])
+
+        for col,id in zip(
+            [
+                self._lane_col,
+                self._sample_col,
+                self._library_name_col,
+                self._platform_unit_col,
+                self._sequencing_center_col,
+                self._platform_col,
+                self._platform_model_col,
+                self._sequencing_date_col
+            ],
+            ["ID","SM","LB","PU","CN","PL","PM","DT"]):
+            if row.get(col):
+                read_group_info.append("%s:%s" % (id,row[col]))
+
+        tmp_dict['read_group']="RG@:%s\\tDS:%s" % ("\\t".join(read_group_info),"|".join(description))
```

Comment:
Should it be `@RG` instead of `RG@`?

Reply:
Good catch. Addressed:
https://github.com/icgc-argo-workflows/argo-modules/blob/e7871dd8ba5a717740a230c2487f3e3de3492f04/modules/icgc-argo-workflows/checkinput/resources/usr/bin/dnaaln.py#L175